Source/WebCore/platform/ios/VideoFullscreenInterfaceAVKit.mm

```diff
@@ -461,31 +450,53 @@ - (void)exitFullScreenAnimated:(BOOL)animated completionHandler:(void (^)(BOOL s
 #define MY_NO_RETURN
 #endif
 
-#if HAVE(AVOBSERVATIONCONTROLLER)
-static const NSTimeInterval startPictureInPictureTimeInterval = 0.5;
+static const NSTimeInterval startPictureInPictureTimeInterval = 1.0 ;
+
+- (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(NSDictionary *)change context:(void *)context
+{
+    if (!_startPictureInPictureTimer)
+        return;
+    [_startPictureInPictureTimer invalidate];
+    _startPictureInPictureTimer = nil;
+
+    [self removeObserver];
+
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [self startPictureInPicture];
+    });
```

Comment:
1 - It is probably worth ASSERTing that `keyPath` is `@"pictureInPicturePossible"`

2 - Do you want to call `startPictureInPicture` if `@"pictureInPicturePossible"` changes to `NO`?

Reply:
2 can't happen: we only register the observer if it's NO to start with.
If it becomes YES, we remove the observer..

Reply:
You remove the observer if any KVO fires, whether the key value changes or not. This change is necessary to work around an AVKit bug, are you certain they won't ever have a bug here?